tests/OrderableRowsThroughVersionedTest.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Symbiote\GridFieldExtensions\Tests;
+
+use ReflectionMethod;
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridFieldConfig_RelationEditor;
+use Symbiote\GridFieldExtensions\Tests\Stub\ThroughDefiner;
+use Symbiote\GridFieldExtensions\Tests\Stub\ThroughIntermediary;
+use Symbiote\GridFieldExtensions\Tests\Stub\ThroughBelongs;
+use Symbiote\GridFieldExtensions\GridFieldOrderableRows;
+
+class OrderableRowsThroughTest extends SapphireTest
+{
+    protected static $fixture_file = 'OrderableRowsThroughTest.yml';
+
+    protected static $extra_dataobjects = [
+        ThroughDefiner::class,
+        ThroughIntermediary::class,
+        ThroughBelongs::class,
+    ];
+
+    protected static $required_extensions = [
+        ThroughDefiner::class => [ Versioned::class ],
+        ThroughIntermediary::class => [ Versioned::class ],
+        ThroughBelongs::class => [ Versioned::class ],
+    ];
+
+    /**
+     * Mostly the same as GridFieldOrderableRowsTest::testReorderItems
+     * but with some Versioned calls & checks mixed in.
+     */
+    public function testReorderingSaves()
+    {
+        $parent = $this->objFromFixture(ThroughDefiner::class, 'DefinerOne');
+        $sortName = 'Sort';
+
+        $orderable = new GridFieldOrderableRows($sortName);
+        $reflection = new ReflectionMethod($orderable, 'executeReorder');
+        $reflection->setAccessible(true);
+
+        $config = new GridFieldConfig_RelationEditor();
+        $config->addComponent($orderable);
+
+        // This test data is versioned - ensure we're all published
+        $parent->publishRecursive();
+        // there should be no difference between stages at this point
+        foreach ($parent->Belongings() as $item) {
+            if ($item->stagesDiffer()) {
+                $this->fail('Unexpected diference found on stages');
+            }
+        }
+
+        $grid = new GridField(
+            'Belongings',
+            'Testing Many Many',
+            $parent->Belongings()->sort($sortName),
+            $config
+        );
+
+        $originalOrder = $parent->Belongings()->sort($sortName)->column('ID');
+        $desiredOrder = [];
+
+        // Make order non-contiguous, and 1-based
+        foreach (array_reverse($originalOrder) as $index => $id) {
+            $desiredOrder[$index * 2 + 1] = $id;
+        }
+
+        $this->assertNotEquals($originalOrder, $desiredOrder);
+
+        $reflection->invoke($orderable, $grid, $desiredOrder);
+
+        $newOrder = $parent->Belongings()->sort($sortName)->map($sortName, 'ID')->toArray();
+
+        $this->assertEquals($desiredOrder, $newOrder);
+
+        // pass forward to testReorderedPublish
+        return $parent;
```

Comment:
I don't think is happening, and if it were I think I'd suggest it'd be better not to couple unit tests together. I see you've declared `testReorderedPublish` as depending on this test, but it doesn't actually use its return value - can you remove

Reply:
Sorry, can't do. It is passed in at line 86, and used on line 90, 98, & 101 (in `testReorderedPublish`). I agree that normally it's not a good idea to string tests along like this, but in this case I feel it's acceptable in that if the first test fails there's no point in testing the second as it truly is a prerequisite (in order to publish one must be able to save) - I'm not aware of any method to fixture version information.

It could also be argued that this could all be part of one big test if this dependency is the case, but I feel it is cleaner this way.

Reply:
Added docblock `@param` for `testReorderedPublish` to compliment the comment above the return there and make it a bit clearer :)

Reply:
Ah I see, the return value is passed into testReorderedPublish as an arg. Ok